equinix/resource_metal_port.go

```diff
@@ -219,7 +222,7 @@ func resourceMetalPortDelete(d *schema.ResourceData, meta interface{}) error {
 			return err
 		}
 		for _, f := range [](func(*ClientPortResource) error){
```

Comment:
Should all of the iterated functions (`portSanityChecks`, `makeDisbond`, `convertToL2`, `makeBond`, `convertToL3`, `updateNativeVlan`, (others?)) accept a `context.Context`? 

```suggestion
		for _, f := range [](func(context.Context, *ClientPortResource) error){
```

`batchVlans` would be modified to return `func(context.Context, *ClientPortResource) error` rather than accept context (as in `batchVlans(ctx, start, false)`).

Reply:
Looking at these helpers, the only ones that would benefit from context are the ones that use the cpr.Client either directly or indirectly. (example https://github.com/equinix/terraform-provider-equinix/blob/a34837fc3b1105075814494ca4edcc52b7f1018d/equinix/port_helpers.go#L213-L216).

Since these are packngo function that don't accept context, your approach seems appropriate. We may want to reconsider this when metal-go is put to use here.

Reply:
> Since these are packngo function that don't accept context, your approach seems appropriate. We may want to reconsider this when metal-go is put to use here.
Yes thats right